ssl/bio_ssl.c

```diff
@@ -101,14 +102,14 @@ static int ssl_read(BIO *b, char *out, int outl)
 
     BIO_clear_retry_flags(b);
 
-    ret = SSL_read(ssl, out, outl);
+    ret = SSL_read_ex(ssl, out, outl, &readbytes);
 
     switch (SSL_get_error(ssl, ret)) {
     case SSL_ERROR_NONE:
-        if (ret <= 0)
+        if (readbytes == 0)
```

Comment:
So are you telling me that `SSL_read_ex()` might return 1 when 0 bytes have been read?  I thought you defined that as a failure...  Now, observe [these lines](https://github.com/openssl/openssl/blob/master/ssl/ssl_lib.c#L2912-L2913) in `SSL_get_error()`...  it seems it will only return `SSL_ERROR_NONE` if `ret > 0`.

Of course, it doesn't hurt...  but it *is* confusing when why is trying to understand the return codes from `SSL_read_ex()`